### PR TITLE
[AMD] Fix draft-decode seqused_k in aiter unified_attention

### PR DESCRIPTION
**Pull request by sogalin**
Under EAGLE draft decode the forward_decode unified_attention branch was passing forward_batch.seq_lens as seqused_k. That value is the target's per-request prefix length and does not include the draft tokens that were committed to the draft KV cache in earlier steps of the same round. As a result draft step 1 / step 2 attended only over the prefix, missing the K/V the draft had just written, which cascaded into worse draft logits and noticeably lower accept length.

Compute the per-row kv length from spec_info.kv_indptr inside _build_unified_page_table_from_spec (or write it in-place into a static cuda-graph buffer), stash it on ForwardMetadata.seqused_k, and route it into the unified_attention call. The non-spec path still uses forward_batch.seq_lens unchanged.

Verified on Kimi-K2.5 MXFP4 TP8 + EAGLE3 with SGLANG_USE_AITER_UNIFIED_ATTN=1 (random 8k in / 1k out, concurrency 4, 32 prompts):

  before: Accept length 3.26, output 675 tok/s
  after : Accept length 3.97, output 800 tok/s  (+18% throughput)

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

**Comment by hubertlu-tw** [comment]
Hi @sogalin,
 
On the Qwen3.5-397B-FP8 workload the patch does not introduce any accuracy or throughput regression. The Kimi-K2.5 MXFP4 EAGLE3 gain from the PR description is a reasonable additional motivation even though we do not reproduce it on this particular setup — the bug class (draft-decode seqused_k ignoring committed draft KV) is real and the fix is narrow and CUDA-graph-safe.

```bash
SGLANG_USE_AITER_UNIFIED_ATTN=1 SGLANG_USE_AITER=1 \
SGLANG_AITER_UNIFIED_VERIFY=1 \
python3 -m sglang.launch_server \
  --model-path /data2/Qwen/Qwen3.5-397B-A17B-FP8/ --tp 8 \
  --attention-backend aiter --trust-remote-code \
  --model-loader-extra-config '{"enable_multithread_load": true}' \
  --watchdog-timeout 1200 --mem-fraction-static 0.9 \
  --host 0.0.0.0 --port 9000 --disable-radix-cache \
  --enable-aiter-allreduce-fusion --max-running-requests 128 \
  --page-size 16 \
  --speculative-algorithm EAGLE --speculative-num-steps 3 \
  --speculative-eagle-topk 1 --speculative-num-draft-tokens 4
```
 
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>


<p style='margin-top:13pt;margin-bottom:6pt;font-family:Calibri;font-size:10.0pt'>Performance
— 8k in / 8k out, 64 prompts, c=16</p>

<div style='direction:ltr'>


Metric | Baseline | Patched | Δ
-- | -- | -- | --
Output token   throughput (tok/s) | 2408.48 | 2431.52 | +0.96%
Accept length | 3.40 | 3.40 | 0
Median TTFT (ms) | 183.78 | 217.91 | +34 ms
Median TPOT (ms) | 5.89 | 5.93 | +0.04 ms
Concurrency actual | 14.83 | 15.08 | —



</div>


</body>

</html>

---

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>


<p style='margin-top:13pt;margin-bottom:6pt;font-family:Calibri;font-size:10.0pt'>Performance
— 8k in / 1k out, 32 prompts, c=4 (PR-author-analogous scenario)</p>

<div style='direction:ltr'>


Metric | Baseline | Patched | Δ
-- | -- | -- | --
Output token   throughput (tok/s) | 782.27 | 786.18 | +0.5%
Accept length | 3.40 | 3.32 | −0.08 (within   noise on 32 prompts)
Median TPOT (ms) | 4.39 | 4.49 | +0.10 ms



</div>


</body>

</html>
